module/documents/advancement/item-grant.mjs

```diff
@@ -107,21 +118,38 @@ export default class ItemGrantAdvancement extends Advancement {
       this.actor.updateSource({items: [item]});
       updates[item._id] = item.flags.dnd5e.sourceId;
     }
-    this.updateSource({"value.added": updates});
+    this.updateSource({[this.storagePath(level)]: updates});
   }
 
   /* -------------------------------------------- */
 
   /** @inheritdoc */
   reverse(level) {
     const items = [];
-    for ( const id of Object.keys(this.value.added ?? {}) ) {
+    const keyPath = this.storagePath(level);
+    for ( const id of Object.keys(foundry.utils.getProperty(this, keyPath) ?? {}) ) {
       const item = this.actor.items.get(id);
       if ( item ) items.push(item.toObject());
       this.actor.items.delete(id);
     }
-    this.updateSource({ "value.-=added": null });
+    this.updateSource({[keyPath.replace(/\.([\w\d]+)$/, ".-=$1")]: null});
```

Comment:
It should work with multiple dots because I used `$` to anchor it to the end of the string, so it will only match a set of letters and numbers from the end leading to the first dot from the right.

Reply:
You are right, I missed that detail, this is fine then.